subprojects/core-api/src/main/java/org/gradle/api/tasks/IgnoreDirectories.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks;
+
+import org.gradle.api.Incubating;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Attached to a task input property to specify that directories should be ignored
+ * when snapshotting inputs.  Files within directories and subdirectories will be
+ * snapshot, but the directories themselves will be ignored (i.e. empty directories
+ * will have no effect on the resulting snapshot).
+ *
+ * <p>This annotation should be attached to the getter method in Java or the property in Groovy.
+ * Annotations on setters or just the field in Java are ignored.</p>
+ *
+ * <ul><li>{@link org.gradle.api.tasks.InputFiles}</li>
+ *
+ * <li>{@link org.gradle.api.tasks.InputDirectory}</li> </ul>
```

Comment:
This comes from modeling it after the `SkipWhenEmpty` javadoc.  I've changed it to explain that these are the types it can be applied to as well as to add the transform property types.